src/library/hashtable.lisp

```diff
@@ -1,69 +1,88 @@
 (in-package #:coalton-library)
 
+#-sbcl (error "Coalton hash tables are only supported on SBCL!")
```

Comment:
Although I think this is definitely a better interface over hash tables, I guess I'm not super comfortable with the change to have the standard library being supported on SBCL. I have looked in to support for different implementations in the past and they seemed to not require major changes, however a few were blocked on a backend rewrite for better codegen ordering. 

I would feel more comfortable merging this if we push for hash tables to be a non-prelude feature of the standard library and if there is a clear push forward on #331 such that the standard library does not sit broken on other implementations for too long.